File: 144. Binary Tree Preorder Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    List<Integer> lst = new ArrayList<Integer>();
    public List<Integer> preorderTraversal(TreeNode root) {
        preorder(root);
        return lst;
    }
    
    public void preorder(TreeNode root) {
        if(root != null) {
            lst.add(root.val);
            preorder(root.left);
            preorder(root.right);
        }
    }
}

File: 144. Binary Tree Preorder Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def preorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        def preorder(root):
            if not root:
                return []
            output = [root.val]
            output += preorder(root.left)
            output += preorder(root.right)
            
            return output
        
        return preorder(root)
